module/zfs/zstd/common/compiler.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2016-present, Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_COMPILER_H
+#define ZSTD_COMPILER_H
```

Comment:
We may need to map these to the correct kernel equivilants and remove dead code.  For example, the `_MSC_VER` blocks.  Assuming `CACHELINE_SIZE=64` is also a little concerning, because we do care about other arches.

Reply:
for the CACHELINE_SIZE which is used for prefetching data i assume 64 bytes is pretty standard. for intel machines. for mips and arm i know its almost 32 bytes. i can make this architecture conditional

Reply:
for the rest i will implement the kernel equivilants.

Reply:
partially done. usually more todo